src/rover/main.cpp

```diff
@@ -0,0 +1,174 @@
+		duration_cast<milliseconds>(end_grab_frame - start_all).count() <<
+		"ms" <<
+	std::endl;
+#endif
+
+
+
+		/*
+		 * 2. Send out packets
+		 * - Send Logging packets
+		 * - Create camera messages and send Camera packets
+		 */
+		network::poll_incoming(&conn);
+		network::Message message;
+		// Receive incoming messages //
+		while(network::dequeue_incoming(&conn, &message)) {
+				switch(message.type) {
+					case network::MessageType::HEARTBEAT:
+					{
+						network::Buffer* outgoing = network::get_outgoing_buffer();
+						network::queue_outgoing(&conn, network::MessageType::HEARTBEAT, outgoing);
+						break;
+					}
+					default:
+						break;
+				}
+		}
+		// THIS IS SAMPLE CODE to send outgoing packets //
+		// Log messages
+		network::Buffer* log_buffer = network::get_outgoing_buffer();
+		char s[] = "[LOG] Hewwo Wowld!";
+		network::LogMessage log_message = {static_cast<uint8_t>(strlen(s)), s};
+		network::serialize(log_buffer, &log_message);
+
+		network::queue_outgoing(&conn, network::MessageType::LOG, log_buffer);
+
+		// Camera Frames
+		//Temporary 'for' loop to use the same frame from session0 for all streams
+		for (unsigned int i = 0; i < 4; i++) {
+			uint8_t num_buffers = (frame_size/CAMERA_MESSAGE_SIZE) + 1;
+			for (unsigned int j = 0; j < num_buffers; j++) {
+				network::Buffer* camera_buffer = network::get_outgoing_buffer();
+				// If we are not at the last buffer
+				uint16_t buffer_size = (j != num_buffers-1)?CAMERA_MESSAGE_SIZE:frame_size%CAMERA_MESSAGE_SIZE;
+				network::CameraMessage message = {
+					static_cast<uint8_t>(i),              // stream_index
+					static_cast<uint16_t>(frame_counter), // frame_index
+					static_cast<uint8_t>(j),              // section_index
+					num_buffers,                          // section_count
+					buffer_size,                          // size
+					nullptr                               // data
+				};
+				message.data = (uint8_t*) malloc(CAMERA_MESSAGE_SIZE);
+				// OPTIMIZE: reuse the same message object to avoid reallocating memory
+				memcpy(message.data, frame_buffer+CAMERA_MESSAGE_SIZE*j, CAMERA_MESSAGE_SIZE);
+				network::serialize(camera_buffer, &message);
+
+				network::queue_outgoing(&conn, network::MessageType::CAMERA, camera_buffer);
+			}
+		}
+
+		network::drain_outgoing(&conn);
+		
+		/// Return all buffers
+		err1 = camera::return_buffer(&session1, frame_buffer);
+
+
+#ifdef DEBUG
+		std::cout << "2" << camera::get_error_string(err1) << std::endl;
+#endif
+#ifdef TIME
+		auto end_all = high_resolution_clock::now();
+		std::cout << 
+			"[LOCAL_TIME] Time to CREATE and SEND all Logging and Camera buffers: " <<
+			duration_cast<milliseconds>(end_all - end_grab_frame).count() <<
+			"ms" <<
+		std::endl;
+#endif
+	}
+
+
+	/// Destruct Rover objects
+	// Close the camera session
+	camera::close(&session0);
+	camera::close(&session1);
+	camera::close(&session2);
+	camera::close(&session3);
+	return 0;
+}
```

Comment:
Again, can't comment below: Do we want a movie file committed to the repo?

Reply:
I thought it would be helpful to give a compatible video file for people to start off with; but since the video spoofing is a little finicky anyways, I've removed it, along with the shell script